### ISSUE TITLE
If omitted all chunks are used?? Could you explain what do you mean?

### ISSUE THREAD

**Issue by pksjce** [webpack/webpack.js.org#123]
_From @wzup on August 1, 2016 9:27_

Could you please elaborate what you actually meant?

**CommonsChunkPlugin** 

At first read this, bold and italic part:

> `options.name` or `options.names` (`string|string[]`): The chunk name of the commons chunk. An existing chunk can be selected by passing a name of an existing chunk. If an array of strings is passed this is equal to invoking the plugin multiple times for each chunk name. **_If omitted and `options.async` or `options.children` is set all chunks are used**_, otherwise `options.filename` is used as chunk name.

Now read this, bold and italic part:

> `options.async` (`boolean|string`): If `true` **_a new async commons chunk is created as child of `options.name`**_ and sibling of `options.chunks`. It is loaded in parallel with options.chunks. It is possible to change the name of the output file by providing the desired string instead of true.

And a question is. 
How can you create a child of something that is omitted, that doesn't exist? 
1. If `options.name` or  `options.names`  is omitted and `options.async` is set
2. If  `options.async` is `true` a new async commons chunk is created as child of `options.name`

What did you actually mean?

_Copied from original issue: webpack/webpack#2827_

**Comment by pksjce** [comment]
_From @sokra on September 1, 2016 19:9_

If
* `name`(s) is not set
* `async` or `children` is true

then

all chunks are used by the CommonChunkPlugin.

The CommonsChunkPlugin iterates over all chunks of the compilation and does it job. It checks for common modules in the children of the chunk and creates an async chunk with the common modules.

**Comment by skipjack** [comment]
@wzup can you take a look at the [new documentation](https://webpack.js.org/plugins/commons-chunk-plugin/) for the `CommonsChunkPlugin`? It's more elaborate and contains a lot of detailed examples. 

If there is we would definitely welcome a PR, I agree the descriptions are a bit hard to follow. I know this is a bit old but better to get a fix in late than never? Also, did @sokra's description help at all? I think maybe a few minor adjustments to the phrasing of the option descriptions could go a long way.

**Comment by skipjack** [comment]
@wzup closing for inactivity but please see the new CommonsChunkPlugin page I linked to above and let us know if the problem persists.
